repair/row_level.cc

```diff
@@ -551,9 +551,16 @@ void repair_writer_impl::create_writer(lw_shared_ptr<repair_writer> w) {
     }
     replica::table& t = _db.local().find_column_family(_schema->id());
     rlogger.debug("repair_writer: keyspace={}, table={}, estimated_partitions={}", w->schema()->ks_name(), w->schema()->cf_name(), w->get_estimated_partitions());
+    auto off_str = is_offstrategy_supported(_reason);
+    // #17384 don't use off-strategy for repair if using tablets. sstables generated will 
+    // be single token range and can just be added to normal sstable set as is, eventually
+    // handled by normal compaction.
+    if (_reason == streaming::stream_reason::repair && t.uses_tablets()) {
```

Comment:
I think `_reason` should not be part of this check. Regardless of the reason, when a table uses tablets, offstrategy is not needed.

@bhalevy what do you think?

Reply:
Does this then in fact mean it can be skipped for the streaming case as well? (`stream_session.cc:89: stream_manager::make_streaming_consumer`)

Reply:
(And also hypothetical, as that part does not have all info at that place etc.. - i.e. maybe refactor further)

Reply:
We don't, streaming can keep using off-strategy, it is vnode only.